oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java

```diff
@@ -202,6 +194,15 @@ public AccessToken refreshAccessToken() throws IOException {
       stsTokenExchangeRequest.setScopes(new ArrayList<>(scopes));
     }
 
+    // If the workforcePoolUserProject is set, we need to pass it to STS via the the internal
+    // options param.
+    if (workforcePoolUserProject != null && !workforcePoolUserProject.isEmpty()) {
```

Comment:
Can you elaborate on this?

Reply:
its a one way to move out this code into workforce credential if we go ahead and add one.

Reply:
@lsirac One simple alternative is to define properties like isWorkForcePoolMode (and use the line above as a get implementation) and isIdentityPoolMode and document the difference. And use those in conditional statements. Otherwise there is only indirect reference to workforce pool.

Reply:
Ok, I added an isWorkforcePoolConfiguration() method with some documentation.